scripts/solc-bin/bytecode_reports_for_modified_binaries.sh

```diff
@@ -0,0 +1,133 @@
+
+cd "$tmp_dir"
+
+git clone https://github.com/ethereum/solc-js.git "$solcjs_dir"
+cd "$solcjs_dir"
+npm install
+rm soljson.js
+
+cd "${solc_bin_dir}/${platform}/"
+echo "Commit range: ${base_ref}..${top_ref}"
+
+modified_release_versions=$(
+    git diff --name-only "${base_ref}" |
+    sed -n -E 's/^[^\/]+\/(solc|soljson)-[0-9a-zA-Z-]+-v([0-9.]+)\+commit\.[0-9a-f]+(.[^.]+)?$/\2/p' |
+    sort -V |
+    uniq
+)
+echo "Release versions modified in the commit range:"
+echo "$modified_release_versions"
+
+# NOTE: We want perform the check when the soljson-* files in bin/ and wasm/ are modified too
+# because in that case the symlinks in emscripten-wasm32/ and emscripten-asmjs/ might remain
+# unchanged but were're assuming that these directories are never directly used as a platform name.
+[[ $platform != bin && $platform != wasm ]] || { echo "Invalid platform name."; exit 1; }
+
+platform_binaries="$(git ls-files "solc-${platform}-v*+commit.*" | sort -V)"
+
+for binary_name in $platform_binaries; do
+    solidity_version=$(echo "$binary_name" | sed -n -E 's/^solc-'"${platform}"'-v([0-9.]+)\+commit.+$/\1/p')
+
+    if echo "$modified_release_versions" | grep -x "$solidity_version"; then
+        echo "Binary ${binary_name} (version ${solidity_version}) matches one of the modified versions."
+
+        work_dir="${tmp_dir}/${binary_name}"
+        mkdir "$work_dir"
+        cd "$work_dir"
+
+        # While bytecode scripts come from the latest compiler, the test files should come from
+        # the Solidity version we're running them against to avoid errors due to breaking syntax changes.
+        git clone --branch "v${solidity_version}" "$solidity_dir" "${work_dir}/solidity/"
+        "${script_dir}/isolate_tests.py" "${work_dir}/solidity/test/"
+
+        if [[ $platform == emscripten-wasm32 ]] || [[ $platform == emscripten-asmjs ]]; then
+            ln -sf "${solc_bin_dir}/${platform}/${binary_name}" "${solcjs_dir}/soljson.js"
+            ln -s "${solcjs_dir}" solc-js
+            cp "${script_dir}/bytecodecompare/prepare_report.js" prepare_report.js
+
+            # shellcheck disable=SC2035
+            ./prepare_report.js --strip-smt-pragmas *.sol > "${report_dir}/report-${binary_name}.txt"
+        else
+            yul_optimizer_flags=()
+            if [[ $solidity_version == 0.6.0 ]] || [[ $solidity_version == 0.6.1 ]]; then
+                yul_optimizer_flags+=(--force-no-optimize-yul)
+            fi
+
+            "${script_dir}/bytecodecompare/prepare_report.py" "${solc_bin_dir}/${platform}/${binary_name}" \
+                --interface cli \
+                --smt-use strip-pragmas \
+                --report-file "${report_dir}/report-${binary_name}.txt" \
+                "${yul_optimizer_flags[@]}"
+        fi
+
+        rm -r "${work_dir}/solidity/"
+    else
+        echo "Binary ${binary_name} (version ${solidity_version}) does not match any modified version. Skipping."
+    fi
+done
+
+echo rm -r "$tmp_dir"
```

Comment:
Echoing it, but not doing it? Is that intentional :-)? Or should that even be trapped or sth like that? Or is keeping partial results desirable?

Reply:
Good catch. That's actually a bug.

I initially tested it locally with `rm`s disabled this way to prevent it from trashing my filesystem in case it was buggy :) I'll remove the `echo`.

Reply:
Fixed.